### PR TITLE
Respect all informational status codes.

### PR DESCRIPTION
**Pull request by anuraaga**
Motivation:

HTTP 102 (WebDAV) is not correctly treated as an informational response

Modification:

Delegate all `1XX` status codes to superclass, not just `100` and `101`.

Result:

Supports WebDAV response.
Removes a huge maintenance [headache](https://github.com/line/armeria/pull/2210) in Armeria which has forked the class for these features

/cc @trustin @minwoox

### CODE REVIEW

File: codec-http/src/test/java/io/netty/handler/codec/http/HttpClientCodecTest.java [hunk 1]
```diff
@@ -332,6 +332,28 @@ public void testWebSocket00Response() {
...
+        assertThat(res.protocolVersion(), sameInstance(HttpVersion.HTTP_1_1));
+        assertThat(res.status(), is(HttpResponseStatus.PROCESSING));
+        HttpContent content = ch.readInbound();
+        // HTTP 102 is not allowed to have content.
+        assertThat(content.content().readableBytes(), is(0));
+        content.release();
+
+        assertThat(ch.finish(), is(false));
+
+        assertThat(ch.readInbound(), is(nullValue()));
```
Comment (normanmaurer) [inline codec-http/src/test/java/io/netty/handler/codec/http/HttpClientCodecTest.java]: this is redundant with the above assert. As if `finish()` returns `false` there will be nothing left to consume

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpClientCodec.java [hunk 2]
```diff
@@ -42,7 +42,7 @@
  *
  * @see HttpServerCodec
  */
-public final class HttpClientCodec extends CombinedChannelDuplexHandler<HttpResponseDecoder, HttpRequestEncoder>
+public class HttpClientCodec extends CombinedChannelDuplexHandler<HttpResponseDecoder, HttpRequestEncoder>
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpClientCodec.java]: I am a bit hesitate to do this... Can you explain what exactly you do when you extend this ?

Comment (anuraaga) [reply codec-http/src/main/java/io/netty/handler/codec/http/HttpClientCodec.java]: Thanks for pushing back - our usage is here

https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java#L634

But this line of code is positively ancient (almost 5 years old). I just tried removing it and didn't find any test failure and suspect we don't actually need to extend the codec. Removed that change.

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@slandelle PTAL as well

**Comment by normanmaurer** [comment]
@anuraaga thanks a lot!

**Comment by normanmaurer** [comment]
@anuraaga also noticed this bug while merging your pr: https://github.com/netty/netty/pull/9721 PTAL

**Pull request merged**